compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -554,6 +554,10 @@ fn run_runtime_cleanup_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
 }
 
 fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+    if body.optimization_disabled {
```

Comment:
I don't think this is actually correct - some of these should run in debug mode. Not sure how best to decide which passes should be omitted

Reply:
This is indeed incorrect. There are required passes that must be run on every MIR for soundness IIRC.

@rust-lang/wg-mir-opt: Anyone willing to provide assistance to how to disable MIR optimizations while only keeping the required passes?

Reply:
Maybe check it against the opt level of the pass. If it runs on opt level 0, then don't disable it

Reply:
From a brief look, there are 3 or 4 passes that need keeping, including marking the mir phase. 
The easiest way would be to compute the list of passes based on the opt level, or tweak the pass manager to compare default opt level with this flag.